src/nanoarrow/nanoarrow_device_cuda.c

```diff
@@ -312,17 +319,26 @@ static ArrowErrorCode ArrowDeviceCudaBufferCopy(struct ArrowDevice* device_src,
   return result;
 }
 
-static ArrowErrorCode ArrowDeviceCudaBufferInit(struct ArrowDevice* device_src,
+static ArrowErrorCode ArrowDeviceCudaBufferCopy(struct ArrowDevice* device_src,
                                                 struct ArrowBufferView src,
                                                 struct ArrowDevice* device_dst,
-                                                struct ArrowBuffer* dst) {
+                                                struct ArrowBufferView dst) {
+  return ArrowDeviceCudaBufferCopyAsync(device_src, src, device_dst, dst,
+                                        NANOARROW_CUDA_DEFAULT_STREAM);
+}
```

Comment:
Same issue here where if this is assumed to be synchronous then this can cause races for other non-blocking streams